### ISSUE TITLE
Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js"

### ISSUE THREAD

**Issue by hoogw** [josdejong/jsoneditor#1208]
I got this warning when build, using "jsoneditor": "^9.1.6",

                     Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js"

full code:

                       parcel build --no-minify --no-cache --public-url ./ arcgis/*.html

                     ⚠️  Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js".
                     ⚠️  Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js".
                     ✨  Built in 8.14s.

**Comment by josdejong** [comment]
How can I reproduce this issue?

**Comment by hoogw** [comment]
how to reproduce issue:




                              npm init

                              npm install jsoneditor


I use parcel.js to bundle.   install parcel.js by: 


                              npm install --save-dev parcel-bundler     


 create a simple html, js file.
in js file, I will import jsoneditor as: 

                                

                             // jsoneditor
                                import 'jsoneditor/dist/jsoneditor.css';   // do not forget CSS !
                                import JSONEditor from "jsoneditor";


Then, run:

                     parcel build --no-minify --no-cache --public-url ./    *.html



Then you will see warning

                          Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js".
                 ⚠️  Could not load source file "0" in source map of "../node_modules/jsoneditor/dist/jsoneditor.min.js".
                 ✨  Built in 8.14s.

This is my package.json, you could have yours, 

                                                     {
                     "name": "ol2",                          
                     "version": "1.0.0",
                     "description": "",
                     "main": "index.js",
                            "scripts": {
                              "test": "echo \"Error: no test specified\" && exit 1",
                              "start": "parcel arcgis/*.html",
                              "build": "parcel build --no-minify --no-cache --public-url ./ arcgis/*.html"
                            },
                            "author": "",
                            "license": "ISC",
                            "dependencies": {
                              "@mdi/font": "^5.8.55",
                              "jquery": "^3.5.1",
                              "json": "^10.0.0",
                             "jsoneditor": "^9.1.6",
                             "ol": "^6.4.3",
                              "ol-layerswitcher": "^3.8.3",
                              "ol-mapbox-style": "^6.3.0",
                              "proj4": "^2.6.3",
                              "terraformer": "^1.0.12"
                            },
                            "devDependencies": {
                             "parcel-bundler": "^1.12.4"
                            }
                          }

**Comment by josdejong** [comment]
I managed to reproduce the issue by following your steps and creating:

file **issue_1208.html**:
```html
<html>
  <head>
    <script src="issue_1208.js"></script>
  </head>
  <body>
    
  </body>
</html>
```

and file **issue_1208.js**

```js
// jsoneditor
import 'jsoneditor/dist/jsoneditor.css';   // do not forget CSS !
import JSONEditor from "jsoneditor";
```


Anyone able to look into why Parcel outputs this warning about the source map?

**Comment by Nikki1993** [comment]
It's not just parcel but webpack as well. I am trying to debug an issue and plugging in `source-map-loader` revealed this 

```
WARNING in ../node_modules/jsoneditor/dist/jsoneditor.min.js
Module Warning (from ../node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '[redacted]/client/node_modules/jsoneditor/dist/0' file: Error: ENOENT: no such file or directory, open '[redacted]/client/node_modules/jsoneditor/dist/0'
```

**Comment by josdejong** [comment]
Thanks for the update @Nikki1993 , that means this is probably not a Parcel specific issue but something generic being wrong with the source map of jsoneditor.

**Comment by erichiller** [comment]
Same issue here, and I am not using parcel.

**Comment by erichiller** [comment]
@josdejong 

Looks like all that has to be done is to change the `main` field in your `package.json` to `./dist/jsoneditor.js`

See my [fix](https://github.com/erichiller/inventory-manager-server/blob/ea18531ff24b06387a97b2ed565608b170ca4261/client/fix-jsoneditor-sourcemap.ps1#L3)

**Comment by josdejong** [comment]
> Looks like all that has to be done is to change the `main` field in your `package.json` to `./dist/jsoneditor.js`

hm, yeah not using the minified version (with source map) but the non-minified version will indeed work around the issue, though it doesn't really solve the issue of course.

**Comment by ahmedbecirevic** [comment]
Any updates with this issue, I still have it in the console, with create-react-app project, React v18

**Comment by josdejong** [comment]
No news. Help solving this would be very welcome.

**Comment by sreenaths** [comment]
I was able to prevent the warning by importing from **index**!

```
import JSONEditor from "jsoneditor/index";
```

**Comment by josdejong** [comment]
🤔 that is interesting.

Thinking aloud here: maybe defining the `exports` object in `package.json` will solve this, let me give that a try.

**Comment by josdejong** [comment]
This is fixed now in `v9.10.0` via #1499
